packages/open-payments/src/client/requests.test.ts

```diff
@@ -1,37 +1,73 @@
 /* eslint-disable @typescript-eslint/no-empty-function */
 import { createAxiosInstance, get } from './requests'
+import { generateKeyPairSync } from 'crypto'
 import nock from 'nock'
 import { mockOpenApiResponseValidators, silentLogger } from '../test/helpers'
 
 describe('requests', (): void => {
   const logger = silentLogger
+  const privateKey = generateKeyPairSync('ed25519').privateKey
+  const keyId = 'myId'
 
   describe('createAxiosInstance', (): void => {
     test('sets timeout properly', async (): Promise<void> => {
       expect(
-        createAxiosInstance({ requestTimeoutMs: 1000 }).defaults.timeout
+        createAxiosInstance({ requestTimeoutMs: 1000, privateKey, keyId })
+          .defaults.timeout
       ).toBe(1000)
     })
     test('sets Content-Type header properly', async (): Promise<void> => {
       expect(
-        createAxiosInstance({ requestTimeoutMs: 0 }).defaults.headers.common[
-          'Content-Type'
-        ]
+        createAxiosInstance({ requestTimeoutMs: 0, privateKey, keyId }).defaults
+          .headers.common['Content-Type']
       ).toBe('application/json')
     })
   })
 
   describe('get', (): void => {
-    const axiosInstance = createAxiosInstance({ requestTimeoutMs: 0 })
+    const axiosInstance = createAxiosInstance({
+      requestTimeoutMs: 0,
+      privateKey,
+      keyId
+    })
     const baseUrl = 'http://localhost:1000'
     const responseValidators = mockOpenApiResponseValidators()
 
     beforeAll(() => {
       jest.spyOn(axiosInstance, 'get')
     })
 
+    afterEach(() => {
+      jest.useRealTimers()
+    })
+
     test('sets headers properly if accessToken provided', async (): Promise<void> => {
-      nock(baseUrl).get('/incoming-payment').reply(200)
+      // https://github.com/nock/nock/issues/2200#issuecomment-1280957462
+      jest
+        .useFakeTimers({
+          doNotFake: [
+            'nextTick',
+            'setImmediate',
+            'clearImmediate',
+            'setInterval',
+            'clearInterval',
+            'setTimeout',
+            'clearTimeout'
+          ]
+        })
+        .setSystemTime(new Date())
+
+      const scope = nock(baseUrl)
+        .matchHeader('Signature', /sig1=:([a-zA-Z0-9+/]){86}==:/)
+        .matchHeader(
+          'Signature-Input',
+          `sig1=("@method" "@target-uri" "authorization");created=${Math.floor(
+            Date.now() / 1000
+          )};keyid="${keyId}";alg="ed25519"`
+        )
+        .get('/incoming-payment')
+        // TODO: verify signature
```

Comment:
Should we move the meat of `auth`'s [signature validation](https://github.com/interledger/rafiki/blob/main/packages/auth/src/signature/middleware.ts) into `open-payments` (after #672 is merged)?

Reply:
I'd say yes.

Reply:
What do you mean by `open-payments`?

Reply:
https://github.com/interledger/rafiki/tree/main/packages/open-payments

Reply:
I'm thinking the "receiver"/RS will also be using `open-payments` at least for the types (why I was reluctant to call `open-payments` a "client" library), so why not also provide httpsig verification?

Reply:
@wilsonianb wouldn't the RS also use it for sig creation when requesting resources at other RS's?

Reply:
Correct, the RS will also use ~`createClient`~`createAuthenticatedClient` to send requests.